src/mscorlib/shared/System/IO/PathInternal.cs

```diff
@@ -117,6 +117,15 @@ internal static bool AreRootsEqual(string first, string second, StringComparison
         internal static string RemoveRelativeSegments(string path, int skip = 0)
         {
             Debug.Assert(skip >= 0);
+
+            // It moves the value of the skip to the second last character of the segment.
```

Comment:
Why? This seems wrong. Really `skip` should be `rootLength` should it not? Saying what this technically does isn't enough. If we need to keep this it should state the "why".

`// Move skip forward to the last character in the segment as ... { reason why }`

Reply:
If we don't subtract one we wont be able to remove the first relative segment. eg case like C:\.\tmp, and C:\..\temp.
Technically RemoveRelativeSegments should do this but it will include rewriting the function from scratch. I am little reluctant to do that as we are end of the release cycle. Subtracting one passes all our tests.
I can open an issue to change this behaviour and can start working on it in a separate PR. But for the time being we should merge this change.